File: project/.sdk/src/cmp/rb/fragment/EntityLoadOp.fragment.rb
require_relative '../utility/struct/voxgig_struct'
require_relative '../core/helpers'

# EJECT-START

  # Load a single EntityName.
  #
  # @param reqmatch [EntityNameLoadMatch, Hash, nil] match criteria (id/query fields);
  #   optional — an entity with no id-like key loads with no match (nil is treated
  #   as an empty match, so client.EntityName.load works with no args).
  # @param ctrl [Object, nil] optional per-call control
  # @return [EntityName, Hash] the loaded EntityName; raises ProjectNameError on failure
  def load(reqmatch = nil, ctrl = nil)
    utility = @_utility
    ctx = utility.make_context.call({
      "opname" => "load",
      "ctrl" => ctrl,
      "match" => @_match,
      "data" => @_data,
      "reqmatch" => reqmatch,
    }, @_entctx)

    _run_op(ctx) do
      if ctx.result
        @_match = ctx.result.resmatch if ctx.result.resmatch
        if ctx.result.resdata
          @_data = ProjectNameHelpers.to_map(VoxgigStruct.clone(ctx.result.resdata)) || {}
        end
      end
    end
  end

# EJECT-END

File: project/.sdk/src/cmp/py/fragment/EntityLoadOp.fragment.py
from utility.voxgig_struct import voxgig_struct as vs
from core import helpers

entity_load_op = None

# EJECT-START

    def load(self, reqmatch=None, ctrl=None) -> EntityName:
        utility = self._utility
        # reqmatch is optional: an entity with no id-like key loads with no
        # match. Treat None as an empty match so client.EntityName().load()
        # works with no args.
        if reqmatch is None:
            reqmatch = {}
        ctx = utility.make_context({
            "opname": "load",
            "ctrl": ctrl,
            "match": self._match,
            "data": self._data,
            "reqmatch": reqmatch,
        }, self._entctx)

        def post_done():
            if ctx.result is not None:
                if ctx.result.resmatch is not None:
                    self._match = ctx.result.resmatch
                if ctx.result.resdata is not None:
                    self._data = helpers.to_map(vs.clone(ctx.result.resdata)) or {}

        return self._run_op(ctx, post_done)

# EJECT-END
